SIPS/sip-23.md

```diff
@@ -0,0 +1,627 @@
+```
+
+##### Example
+
+```typescript jsx
+<Image src="<svg>...</svg>" alt="My image" />
+```
+
+#### Input
+
+The `Input` component renders an input element. It accepts a `name` prop which
+is a string representing the input name, a `type` prop which is a string
+representing the input type, and an optional `value` prop which is a string
+representing the input value. The component also accepts an optional
+`placeholder` prop which is a string representing the input placeholder, and an
+optional `onChange` prop which is a function to be called when the input value
+changes.
+
+The `type` MUST be one of `'text'`, `'password'`, or `'number'`.
+
+```typescript
+type InputProps = {
+  name: string;
+  type: 'text' | 'password' | 'number';
+  value?: string | undefined;
+  placeholder?: string | undefined;
+  onChange?: ((value: string) => void) | undefined;
+};
+
+const Input: SnapComponent<InputProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Input name="name" type="text" placeholder="Enter your name" onChange={handleChange} />
+```
+
+#### Italic
+
+The `Italic` component renders a text element in italic. It accepts a `children`
+prop which can be a `string` or an array of `string`s. The component MUST be
+used inside a `Text` component.
+
+```typescript
+type ItalicProps = {
+  children: string | string[];
+};
+
+const Italic: SnapComponent<ItalicProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Text>
+  Hello, <Italic>world</Italic>!
+</Text>
+```
+
+#### Link
+
+The `Link` component renders a link element. It accepts a `children` prop which
+can be a `string` or an array of `string`s, and an `href` prop which is a string
+representing the link URL.
+
+The `href` MUST be an absolute URL. Relative URLs are not supported. It MUST
+start with `https://` or `mailto:`.
+
+```typescript
+type LinkProps = {
+  children: string | string[];
+  href: string;
+};
+
+const Link: SnapComponent<LinkProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Link href="https://example.com">Visit example.com</Link>
+```
+
+#### Row
+
+The `Row` component renders a key-value pair element. It accepts a `label` prop
+which is a string representing the key, and a `children` prop which can be an
+address, image, or text element.
+
+In addition, the component accepts an optional `variant` prop which can be
+`'default'`, `'warning'`, or `'error'`. The default value is `'default'`.
+
+```typescript
+type RowProps = {
+  label: string;
+  children: AddressElement | ImageElement | TextElement;
+  variant?: 'default' | 'warning' | 'error';
+};
+
+const Row: SnapComponent<RowProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Row label="Address">
+  <Address address="0x4bbeEB066eD09B7AEd07bF39EEe0460DFa261520" />
+</Row>
+```
+
+#### Spinner
+
+The `Spinner` component renders a spinner element. It does not accept any props.
+
+```typescript
+const Spinner: SnapComponent;
+```
+
+##### Example
+
+```typescript jsx
+<Spinner />
+```
+
+#### Stack
```

Comment:
Wondering if `Stack` is a confusing name. In Chakra they have the notion of `HStack` and `VStack`. The default behaviour for `Stack` is similar to `Box`.

Reply:
Maybe. We could just use `Box` for all things alignment, horizontal and vertical?

Reply:
Yeah, that's an option too. Whatever minimizes the confusion IMO